crates/bevy_color/src/color_ops.rs

```diff
@@ -97,6 +99,26 @@ pub trait ClampColor: Sized {
     fn is_within_bounds(&self) -> bool;
 }
 
+/// Trait with methods for converting colors to non-color types
+pub trait Convert {
+    /// Convert to an f32 array
+    fn to_f32_array(self) -> [f32; 4];
+    /// Convert to an f32 array without the alpha value
+    fn to_alphaless_array(self) -> [f32; 3];
+    /// Convert to a Vec4
+    fn to_vec4(self) -> Vec4;
+    /// Convert to a Vec3
+    fn to_vec3(self) -> Vec3;
+    /// Convert from an f32 array
+    fn from_array(color: [f32; 4]) -> Self;
+    /// Convert from an f32 array without the alpha value
+    fn from_alphaless_array(color: [f32; 3]) -> Self;
+    /// Convert from a Vec4
+    fn from_vec4(color: Vec4) -> Self;
+    /// Convert from a Vec3
+    fn from_vec3(color: Vec3) -> Self;
```

Comment:
```suggestion
    /// Convert to an f32 array
    fn as_array_f32(self) -> [f32; 4];
    /// Convert to an f32 array without the alpha value
    fn as_array_f32_without_alpha(self) -> [f32; 3];
    /// Convert to a Vec4
    fn as_vec4(self) -> Vec4;
    /// Convert to a Vec3
    fn as_vec3(self) -> Vec3;
    /// Convert from an f32 array
    fn from_array_f32(color: [f32; 4]) -> Self;
    /// Convert from an f32 array without the alpha value
    fn from_array_f32_without_alpha(color: [f32; 3]) -> Self;
    /// Convert from a Vec4
    fn from_vec4(color: Vec4) -> Self;
    /// Convert from a Vec3
    fn from_vec3(color: Vec3) -> Self;
```

I personally don't like the word "_alphaless_", and I would prefer that these methods have the same prefix for the sake of nicer intellisense suggestions. Likewise for the associated `from_x` methods too. Also, I believe `as_x` is the more standard way of performing these conversions ([`as_slice`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.as_slice) in the standard library, and in Bevy there are more definitions named `as_x` than `to_x`).

Reply:
`_no_alpha`?

Reply:
agree with the rename except for the to as change. link below shows rusts naming conventions

https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv